server/consumer.go

```diff
@@ -708,6 +709,12 @@ func (o *consumer) setLeader(isLeader bool) {
 			if o.active = <-o.inch; !o.active {
 				// Check gateways in case they are enabled.
 				o.active = s.hasGatewayInterest(o.acc.Name, o.cfg.DeliverSubject)
+				if o.active {
+					// There is no local interest, but there is GW interest, we
+					// will watch for interest disappearing.
+					// TODO: may need to revisit...
+					o.gwdtmr = time.AfterFunc(o.dthresh, func() { o.watchGWinterest() })
```

Comment:
Could this timer already be running?
Should we call stopAndClearTimer(&o.gwdtmr) first?